tf_quant_finance/volatility/heston_approximation_test.py

```diff
@@ -0,0 +1,154 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Heston Price method."""
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+from absl.testing import parameterized
+from scipy import integrate
```

Comment:
Please avoid using scipy as it fails some of the internal tests

Reply:
ok, there may be other python packages that can do that

Reply:
We had some issues with scipy before and made a decision to not include it as a dependency unless it is needed inside the library itself. 

The test should be for the code inside our library and not for scipy. You should use scipy to get the expected result and test the output against that. See, e.g., conjugate gradient test:
https://github.com/google/tf-quant-finance/blob/03856c3d696a8f03073309fa7aa1c363e99b9f1a/tf_quant_finance/math/optimizer/conjugate_gradient_test.py#L175

Here the minimum value was computed using scipy but it is not computed inside the test.